gradio/blocks.py

```diff
@@ -3033,3 +3065,30 @@ def get_event_targets(
             event = getattr(block, event_name)
             target_events.append(event)
         return target_events
+
+    @document()
+    def route(self, name: str, path: str | None = None):
+        """
+        Adds a new page to the Blocks app.
+        Parameters:
+            name: The name of the page as it appears in the nav bar.
+            path: The url path of the page. If not provided, it is generated from the name.
+        Example:
+            with gr.Blocks() as demo:
+                name = gr.Textbox(label="Name")
+                ...
+            with demo.route("Test", "/test"):
+                num = gr.Number()
+                ...
+        """
+        if path:
```

Comment:
I would probably do some basic validation to ensure that the path does not contain invalid characters such as "?"

Reply:
Also make sure that a user doesn't accidentally reuse the same route for multiple pages in their Gradio app

Reply:
Currently, if you reuse the same path, something funny happens:

<img width="1136" alt="image" src="https://github.com/user-attachments/assets/c3a2f409-f31f-49f1-a0ed-e21e62d50636" />